modules/gapi/include/opencv2/gapi/infer.hpp

```diff
@@ -254,6 +295,36 @@ typename Net::Result infer(Args&&... args) {
     return GInfer<Net>::on(std::forward<Args>(args)...);
 }
 
+struct Generic { };
+
+template<typename Net, typename... Args>
+typename std::enable_if<std::is_same<Net, Generic>::value, GInferOutputs>::type
+infer(const std::string& name, const GInferInputs& inputs)
+{
+    std::vector<GArg> input_args;
+    std::vector<std::string> input_names;
+
+    const auto& blobs = inputs.getBlobs();
+    for (auto&& p : blobs)
+    {
+        input_names.push_back(p.first);
+        input_args.emplace_back(p.second);
+    }
+
+    GKinds kinds(blobs.size(), cv::detail::OpaqueKind::CV_MAT);
+    auto call = std::make_shared<cv::GCall>(GKernel{
+                GInferBase::id(),
+                name,
+                GInferBase::getOutMeta,
+                {}, // outShape will be filled later
+                std::move(kinds)
+            });
+
+    call->setArgs(std::move(input_args));
+    call->params() = InOutInfo{input_names, {}};
```

Comment:
oh

Reply:
modifiable fields are not very ok

Reply:
But it's better than: `getParams()` -> `params.output_names.push_back(name)` -> `setParams()`, isn't it ?

Reply:
In this particular case it is just `setParams(...)`.

Reply:
Yes, but what about updating params ?

Reply:
Maybe `unique_ptr` is more suitable there